### PR TITLE
fix: update make help text

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Example of before/after below

```bash
[nathan@nathan-redhat instructlab (make-help)]$ make help

Usage:
  make <target>

General
  help             Display this help.

Build
  images           Get the current controller, set the path, and build the Containerfile image. (auto-detect the compatible controller)
  cuda             Build container for NVidia CUDA
  hpu              Build container for Intel Gaudi HPU
  rocm             Build container for AMD ROCm (auto-detect the ROCm GPU arch)
  rocm-gfx1100     Build container for AMD ROCm RDNA3 (Radeon RX 7000 series)
  rocm-gfx1030     Build container for AMD ROCm RDNA2 (Radeon RX 6000 series)
  rocm-gfx90a      Build container for AMD ROCm gfx90a (untested)
  rocm-gfx908      Build container for AMD ROCm gfx908 (untested)
  rocm-gfx906      Build container for AMD ROCm gfx906 (untested)
  rocm-gfx900      Build container for AMD ROCm gfx900 (untested)
  toolbox-rocm     Create AMD ROCm toolbox from container
  toolbox-rm       Stop and remove toolbox container

Development
  tests            Run tox -e unit against code
  verify           Run tox -e fmt,lint,spellcheck against code
  docs             Run tox -e docs against code

Linting
  md-lint          Lint markdown files
[nathan@nathan-redhat instructlab (make-help)]$ make help

Usage:
  make <target>

General
  help             Display this help.

Build
  images           Get the current controller, set the path, and build the Containerfile image. (auto-detect the compatible controller)
  cuda             Build container for NVidia CUDA
  hpu              Build container for Intel Gaudi HPU
  rocm             Build container for AMD ROCm (auto-detect the ROCm GPU arch)
  rocm-gfx1100     Build container for AMD ROCm RDNA3 (Radeon RX 7000 series)
  rocm-gfx1030     Build container for AMD ROCm RDNA2 (Radeon RX 6000 series)
  rocm-gfx90a      Build container for AMD ROCm gfx90a (untested)
  rocm-gfx908      Build container for AMD ROCm gfx908 (untested)
  rocm-gfx906      Build container for AMD ROCm gfx906 (untested)
  rocm-gfx900      Build container for AMD ROCm gfx900 (untested)
  toolbox-rocm     Create AMD ROCm toolbox from container
  toolbox-rm       Stop and remove toolbox container

Development
  tests            Run tox -e unit against code
  verify           Run tox -e fmt,lint,spellcheck against code
  spellcheck-sort  Sort spellcheck directory
  docs             Run tox -e docs against code
  md-lint          Lint markdown files
```

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
I rebased this on top of https://github.com/instructlab/instructlab/pull/1156 by @leseb and added some additional text there
```bash
Development
  tests            Run tox -e unit against code
  verify           Run tox -e fmt,lint,spellcheck against code
  spellcheck-sort  Sort spellcheck directory
  docs             Run tox -e docs against code
  md-lint          Lint markdown files

Validation
  check-tox        ensure tox is installed
  check-toolbox    ensure toolbox is installed
  check-engine     ensure the container engine is installed
  check-rocm       ensure rocm is installed
```

**Review by russellb — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -191,3 +172,21 @@ endif
 md-lint: check-engine ## Lint markdown files
 	$(ECHO_PREFIX) printf "  %-12s ./...\n" "[MD LINT]"
 	$(CMD_PREFIX) $(CENGINE) run --rm -v $(CURDIR):/workdir --security-opt label=disable docker.io/davidanson/markdownlint-cli2:v0.12.1 > /dev/null
+
+##@ Validation
```
Comment (leseb) [inline Makefile:176]: This was a mistake, these are internal functions, we don't want to expose them IMO.

**Review by leseb — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (leseb) [reply Makefile:176]: So let's remove this line.

Comment (nathan-weinberg) [reply Makefile:176]: Hmmm any reason we shouldn't have them documented though @leseb? Can you explain the motivation for not?

Comment (leseb) [reply Makefile:176]: Well, these are purely internal functions for the make code to operate correctly, like `_func()` in Python. They are not documented or exposed. Additionally, users are not supposed to use them, they could but it's meaningless, almost a distraction to the purpose of our Make file IMO.

Comment (nathan-weinberg) [reply Makefile:176]: Sounds good guys, will do so in a followup commit

**Review by russellb — APPROVED** [review]

**Review by leseb — APPROVED** [review]

**Review by alimaredia — APPROVED** [review]

**Pull request merged**